File: src/main/java/in/dragonbra/javasteam/steam/handlers/steamfriends/Friend.java
package in.dragonbra.javasteam.steam.handlers.steamfriends;

import in.dragonbra.javasteam.enums.EFriendRelationship;
import in.dragonbra.javasteam.protobufs.steamclient.SteammessagesClientserverFriends.CMsgClientFriendsList;
import in.dragonbra.javasteam.types.SteamID;

/**
 * Represents a single friend entry in a client's friendlist.
 */
public class Friend {
    private SteamID steamID;

    private EFriendRelationship relationship;

    public Friend(CMsgClientFriendsList.Friend friend) {
        steamID = new SteamID(friend.getUlfriendid());
        relationship = EFriendRelationship.from(friend.getEfriendrelationship());
    }

    public SteamID getSteamID() {
        return steamID;
    }

    public EFriendRelationship getRelationship() {
        return relationship;
    }
}

File: src/main/java/in/dragonbra/javasteam/steam/handlers/steamfriends/Friend.kt
package `in`.dragonbra.javasteam.steam.handlers.steamfriends

import `in`.dragonbra.javasteam.enums.EFriendRelationship
import `in`.dragonbra.javasteam.protobufs.steamclient.SteammessagesClientserverFriends.CMsgClientFriendsList
import `in`.dragonbra.javasteam.types.SteamID

/**
 * Represents a single friend entry in a client's friendlist.
 */
class Friend(friend: CMsgClientFriendsList.Friend) {

    /**
     * Gets the [SteamID] of the friend.
     */
    val steamID: SteamID = SteamID(friend.ulfriendid)

    /**
     * Gets the relationship to this friend.
     */
    val relationship: EFriendRelationship = EFriendRelationship.from(friend.efriendrelationship)
}
